src/content/content-safari.js

```diff
@@ -8,11 +8,13 @@ import Package from '../../package.json'
 
 let uiController = null
 let state = null
+let initialized = false
 
 /**
  * State request processing function.
  */
-let handleStateRequest = function handleStateRequest (message) {
+let handleStateRequest = async function handleStateRequest (message) {
+  console.log(`State request received`, message)
   if (!uiController) {
```

Comment:
Let's leave ourselves some breadcrumbs here and add some comments which explain why we wait to create and initialize the UIController until the first state request is received in Safari (i.e. that the content gets injected into every page , etc.)

Reply:
That's a great idea! Added an explanation of why are we doing it this way